opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -402,6 +440,42 @@ impl BatchLogProcessor {
                 );
                 let mut last_export_time = Instant::now();
                 let mut logs = Vec::with_capacity(config.max_export_batch_size);
+                let current_batch_size = current_batch_size_for_thread;
+
+                // This method gets upto `max_export_batch_size` amount of logs from the channel and exports them.
+                // It returns the result of the export operation.
+                // It expects the logs vec to be empty when it's called.
```

Comment:
we can get rid of `logs` vec and instead create a fresh array and pass its slice to export method.?

Reply:
For using array, we need to know the size of the array beforehand which we don't because we don't know how many items are present in the channel. It might be less than `max_export_batch_size`.

Reply:
we can allocate an array of max_export_batch_size, and slice it at the required length. Anyway, this is not a blocker.

Reply:
Always stack allocating a big array even to export very few logs seems wasteful.

What's the benefit of using an array instead of a `vec` here? We are reusing the same `vec` for each export anyway so allocation isn't a concern.